### ISSUE TITLE
Plugins install broken – only 2 plugins (OneSignal and Twilio) displayed

### ISSUE THREAD

**Issue by onemarshall** [mautic/mautic#6191]
**Please DO NOT report security vulnerabilities here. Send them to security@mautic.com instead.**

What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Y
| Feature request? | 
| Enhancement? | 

## Description:
When completing a fresh install and navigating to plugins page, there is only two plug-ins (OneSignal and Twilio) displayed. Also the 'Install/Upgrade Plugins', button does not work when clicked. On clicking OneSignal or Twilio  the modal displays with 'loading...'   but nothing happens!

![screen shot 2018-06-11 at 16 07 14](https://user-images.githubusercontent.com/59648/41240088-9ab261ea-6d91-11e8-8a92-70e579c94523.png)

## If a bug:

| Q   | A
| --- | ---
| Mautic version |  v2.13.1
| PHP version |  7.1.17

### Steps to reproduce:
1.  Download and install
2. go to Settings > Plugins
 
### Log errors: 

[2018-06-11 12:24:52] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []
[2018-06-11 12:25:51] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []
[2018-06-11 12:26:16] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []
[2018-06-11 12:26:18] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []
[2018-06-11 12:26:57] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []
[2018-06-11 12:27:20] mautic.NOTICE: PHP Notice - A non well formed numeric value encountered - in file /home/rob/sites/_apps/mautic/public/live/app/bundles/InstallBundle/Configurator/Step/CheckStep.php - at line 370 [] []

**Comment by Flavio1998** [comment]
On the right side is a button which says "Install/Upgrade Plugins". Click that and many more Plugins will be displayed.

**Comment by RobMarshall** [comment]
When I click the 'Install/Upgrade Plugins' button nothing happens, it just show's the tooltip!

**Comment by spatialy** [comment]
same here ... mautic in docker over nginx/php7-fpm

**Comment by heathdutton** [comment]
Odd... can you go to /s/plugins/reload ?

**Comment by onemarshall** [comment]
Hi, no, I get a '403 Forbidden nginx' error when viewing /s/plugins/reload url

**Comment by Shaz3e** [comment]
@Flavio1998 said click and all plugins appeared instantly. thanks

**Comment by onemarshall** [comment]
> @Flavio1998 said click and all plugins appeared instantly. thanks

Sorry, but I am not sure what you mean?

**Comment by virgilwashere** [comment]
> > @Flavio1998 said click and all plugins appeared instantly. thanks
> 
> Sorry, but I am not sure what you mean?

He means click on `Install/Upgrade Plugins` button on Plugins page.
Which is also achieved via either of:
- Browsing to `https://your.mautic.site/s/plugins/reload`
- Running `sudo -H -u www-data php ./app/console mautic:plugins:reload` on command line

Have you cleared Mautic cache recently?

Remove the directory before running `cache:clear`
`cd /path/to/mautic`
`rm -rf ./app/cache`
`sudo -H -u www-data php ./app/console -vvv cache:clear --no-warmup --no-interaction`


Regarding your nginx error, do you by chance have this in your `mautic-site.conf`

```
	# Deny everything else in /addons or /plugins folder except Assets folder
    location ~ /(addons|plugins)/.*/Assets/ {
        allow all;
        access_log off;
    }
    location ~ /(addons|plugins)/ { deny all; }
```

Change the last line of that block to:
```
    # Add ^ to anchor regex
    location ~ ^/(addons|plugins)/ { deny all; }
```

**Comment by onemarshall** [comment]
Hi virgilwashere,

I updated my site.conf file with your code and I was able to see all the plugins.

> Change the last line of that block to:
> 
> ```
>     # Add ^ to anchor regex
>     location ~ ^/(addons|plugins)/ { deny all; }
> ```

Thank you your help!

**Comment by Shaz3e** [comment]
> > > @Flavio1998 said click and all plugins appeared instantly. thanks
> > 
> > 
> > Sorry, but I am not sure what you mean?
> 
> He means click on `Install/Upgrade Plugins` button on Plugins page.
> Which is also achieved via either of:
> 
> * Browsing to `https://your.mautic.site/s/plugins/reload`
> * Running `sudo -H -u www-data php ./app/console mautic:plugins:reload` on command line
> 
> Have you cleared Mautic cache recently?
> 
> Remove the directory before running `cache:clear`
> `cd /path/to/mautic`
> `rm -rf ./app/cache`
> `sudo -H -u www-data php ./app/console -vvv cache:clear --no-warmup --no-interaction`
> 
> Regarding your nginx error, do you by chance have this in your `mautic-site.conf`
> 
> ```
> 	# Deny everything else in /addons or /plugins folder except Assets folder
>     location ~ /(addons|plugins)/.*/Assets/ {
>         allow all;
>         access_log off;
>     }
>     location ~ /(addons|plugins)/ { deny all; }
> ```
> Change the last line of that block to:
> 
> ```
>     # Add ^ to anchor regex
>     location ~ ^/(addons|plugins)/ { deny all; }
> ```

Hey Thanks for explanation I was in hurry so just wrote one liner

**Comment by Puspendert** [comment]
> @Flavio1998 said click and all plugins appeared instantly. thanks
> Remove the directory before running `cache:clear`
> `cd /path/to/mautic`
> `rm -rf ./app/cache`
> `sudo -H -u www-data php ./app/console -vvv cache:clear --no-warmup --no-interaction`

Note that the `console` file is not in `/app` but now it's inside `/bin`

**Comment by D1XII** [comment]
Sorry, not a master of code here. Still having this issue. Can anyone help?

1. Cleared the cache via ftp (deleted var>cache)
2. Tried clicking "Install / Upgrade Plugins"
3. Tried navigating directly to: #/plugins/reload (received 500 Internal Server Error)
4. @virgilwashere Tried searching for this file "mautic-site.conf" - I don't have it 

I guess I am running branch 3, but I don't know where to confirm that. I changed the name of the downloaded file which probably had version information.
